src/representation_theory/basic.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import algebra.module.basic
+import data.equiv.module
+import algebra.module.linear_map
+import algebra.monoid_algebra.basic
+import linear_algebra.trace
+
+/-!
+# Monoid representations
+
+This file introduces monoid representations and their characters and proves basic lemmas about them,
+including equivalences between different definitions of representations.
+
+## Main definitions
+
+  * `as_monoid_hom`
+  * `as_module`
+  * `rep_space`
+  * `character`
+
+## Implementation notes
+
+A representation of a monoid `G` over a commutative semiring `k` is implemented as a `k`-module `V`
+together with a `distrib_mul_action G V` instance and a `smul_comm_class G k V` instance.
+
+Alternatively, one can use a monoid homomorphism `G →* (V →ₗ[k] V)`. The definitions `as_monoid_hom`
+and `rep_space` allow to go back and forth between these two definitions.
+-/
+
+open monoid_algebra
+
+namespace representation
+
+section
+variables (k G V : Type*) [comm_semiring k] [monoid G] [add_comm_monoid V]
+variables [module k V] [distrib_mul_action G V] [smul_comm_class G k V]
+
+/--
+A k-linear representation of G on V can be thought of as
+a multiplicative map from G into the k-linear endomorphisms of V.
+-/
+def as_monoid_hom : G →* (V →ₗ[k] V) :=
+{ to_fun := distrib_mul_action.to_linear_map k V,
+  map_one' := by { ext, simp, },
+  map_mul' := λ g g', by { ext, simp [mul_smul], }, }
```

Comment:
This exists already, it's [`distrib_mul_action.to_module_End`](https://leanprover-community.github.io/mathlib_docs/algebra/module/linear_map.html#distrib_mul_action.to_module_End)`

Reply:
Oh thanks, I removed it and replaced it by `to_module_End` where it's used.